File: docs-examples/example-config-java/src/test/java/io/micronaut/configuration/openapi/docs/config/TagsGenerationTest.java
package io.micronaut.configuration.openapi.docs.config;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TagsGenerationTest {

    @Test
    void tagsAreGeneratedFromTheClassName() {
        var openApi = OpenApiSpec.load("hello-world-v1.1.yml");

        var tag = openApi.getTags().get(0);
        assertEquals("user-operations", tag.getName());
        assertEquals("User main operations.", tag.getDescription());
        var paths = openApi.getPaths();
        assertEquals(List.of("user-operations"), paths.get("/read/{id}").getGet().getTags());
        assertEquals("read", paths.get("/read/{id}").getGet().getOperationId());
        assertEquals(List.of("user-operations"), paths.get("/save/{id}").getPost().getTags());
        assertEquals("save2", paths.get("/save/{id}").getPost().getOperationId());
        assertEquals(List.of("user-operations"), paths.get("/save").getPost().getTags());
        assertEquals("save", paths.get("/save").getPost().getOperationId());
    }
}

File: docs-examples/example-config-python/src/test/python/micronaut/configuration/openapi/docs/config/TagsGenerationTest.py
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from . import OpenApiSpec


@MicronautTest
class TagsGenerationTest:

    @Test
    def test_tags_are_generated_from_the_class_name(self):
        open_api = OpenApiSpec.load("hello-world-v1.1.yml")

        tag = open_api.getTags().get(0)
        assert tag.getName() == "user-operations"
        assert tag.getDescription() == "User main operations."
        paths = open_api.getPaths()
        assert list(paths.get("/read/{id}").getGet().getTags()) == ["user-operations"]
        assert paths.get("/read/{id}").getGet().getOperationId() == "read"
        assert list(paths.get("/save/{id}").getPost().getTags()) == ["user-operations"]
        assert paths.get("/save/{id}").getPost().getOperationId() == "save2"
        assert list(paths.get("/save").getPost().getTags()) == ["user-operations"]
        assert paths.get("/save").getPost().getOperationId() == "save"
